File: VeraPelle_front_standalone/src/app/core/models/cart-item-dto.ts
import { ProductDTO } from './product-dto';
import { ColorDTO } from './color-dto';

export interface CartItemDTO {
  id: number;
  product: ProductDTO;
  quantity: number;
  subtotal: number;
  thumbnailUrl: string | null;

  // colore selezionato
  selectedColor: ColorDTO;

  // stock residuo della variante selezionata
  variantStockQuantity: number;
}

File: VeraPelle/src/main/java/psw/verapelle/DTO/CartItemDTO.java
package psw.verapelle.DTO;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
public class CartItemDTO {
    private Long id;
    private ProductDTO product;
    private int quantity;
    private double subtotal;
    private String thumbnailUrl;
    private ColorDTO selectedColor;

    /** Stock residuo della variante selezionata */
    private int variantStockQuantity;
}
